Frameworks/CoreText/CTFrame.mm

```diff
@@ -93,74 +82,74 @@ CFDictionaryRef CTFrameGetFrameAttributes(CTFrameRef frame) {
  @Status Interoperable
 */
 CFArrayRef CTFrameGetLines(CTFrameRef frame) {
-    return frame ? static_cast<CFArrayRef>(static_cast<_CTFrame*>(frame)->_lines.get()) : nil;
+    RETURN_NULL_IF(!frame);
+    return frame->_lines;
 }
 
 /**
  @Status Interoperable
 */
-void CTFrameGetLineOrigins(CTFrameRef frameRef, CFRange range, CGPoint origins[]) {
-    _CTFrame* frame = static_cast<_CTFrame*>(frameRef);
-    if (frame) {
-        _boundedCopy(range, frame->_lineOrigins.size(), frame->_lineOrigins.data(), origins);
-    }
+void CTFrameGetLineOrigins(CTFrameRef frame, CFRange range, CGPoint origins[]) {
+    RETURN_IF(!frame);
+    _boundedCopy(range, frame->_lineOrigins.size(), frame->_lineOrigins.data(), origins);
 }
 
 /**
  @Status Interoperable
 */
-void CTFrameDraw(CTFrameRef frameRef, CGContextRef ctx) {
-    _CTFrame* frame = static_cast<_CTFrame*>(frameRef);
-    if (frame && ctx) {
-        std::vector<GlyphRunData> runs;
-
-        for (size_t i = 0; i < frame->_lineOrigins.size() && (frame->_lineOrigins[i].y < frame->_frameRect.size.height); ++i) {
-            _CTLine* line = static_cast<_CTLine*>([frame->_lines objectAtIndex:i]);
-            CGPoint relativePosition = frame->_lineOrigins[i];
-            for (size_t j = 0; j < [line->_runs count]; ++j) {
-                _CTRun* curRun = [line->_runs objectAtIndex:j];
-                if (j > 0) {
-                    // Adjusts x position relative to the last run drawn
-                    relativePosition.x += curRun->_relativeXOffset;
-                }
-                runs.emplace_back(GlyphRunData{ &curRun->_dwriteGlyphRun, relativePosition, (CFDictionaryRef)curRun->_attributes.get() });
+void CTFrameDraw(CTFrameRef frame, CGContextRef ctx) {
+    RETURN_IF(!ctx);
+    RETURN_IF(!frame);
+
+    std::vector<GlyphRunData> runs;
+
+    for (size_t i = 0; i < frame->_lineOrigins.size() && (frame->_lineOrigins[i].y < frame->_frameRect.size.height); ++i) {
+        _CTLine* line = static_cast<_CTLine*>(CFArrayGetValueAtIndex(frame->_lines, i));
+
+        CGPoint relativePosition = frame->_lineOrigins[i];
+
+        for (size_t j = 0; j < [line->_runs count]; ++j) {
```

Comment:
also when you make it private, to track down all these places. well it really goes off ;)